[Flink AI Flow][Document] Improve OPS API

SteNicholas opened the pull request:
## What is the purpose of the change

The comments of the API in `ops.py` don't follow the standard of the API document, which should be improved.


## Brief change log

  - *Update the comments of the API in `ops.py` to follow the standard of the API document.*

## Verifying this change

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -37,15 +37,19 @@ def read_dataset(dataset_info: Union[DatasetMeta, Text, int],
-    :param name: Name of the read_dataset operator.
-    :return: Channel: data output channel.
+    Reads the dataset through the user-defined processor based on the metadata of dataset. The processor reads dataset
+    from external system and generates the :class:`~ai_flow.meta.ai_graph.ai_node.ReadDatasetNode` with
```
full class name should be fixed:
```
:class:`~ai_flow.meta.ai_graph.ai_node.ReadDatasetNode`" -> :class:`~ai_flow.ai_graph.ai_node.ReadDatasetNode`
```

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -37,15 +37,19 @@ def read_dataset(dataset_info: Union[DatasetMeta, Text, int],
+    from external system and generates the :class:`~ai_flow.meta.ai_graph.ai_node.ReadDatasetNode` with
+    :class:`~ai_flow.meta.dataset_meta.DatasetMeta`.
+
+    :param dataset_info: Information about the dataset read. The type of dataset information could be
```
Maybe  `Information about the dataset read by the processor" is easier to be understood?

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -37,15 +37,19 @@ def read_dataset(dataset_info: Union[DatasetMeta, Text, int],
+                                      :class:`~ai_flow.meta.dataset_meta.DatasetMeta` or Text or int. The `dataset_info`
+                                      means the name of dataset when the type is Text and it means the id of dataset
+                                      when the type is int, which gets the metadata of dataset by the id.or name of
+                                      dataset at default.
```
Current description is fine but I think it can be more concise like this:

`or Text(i.e. the name of dataset) or int(i.e. the id of dataset). If the type is Text or int, this method will get the metadata of dataset by the name or id of dataset by default.`

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -37,15 +37,19 @@ def read_dataset(dataset_info: Union[DatasetMeta, Text, int],
+                                      dataset at default.
+    :param read_dataset_processor: The user-defined processor composed of the logic that reads dataset.
+    :param name: Name of the reading dataset operator represent for the
+                            :class:`~ai_flow.meta.ai_graph.ai_node.ReadDatasetNode`.
```
full class name should be fixed:
```
:class:`~ai_flow.meta.ai_graph.ai_node.ReadDatasetNode`" -> :class:`~ai_flow.ai_graph.ai_node.ReadDataNode
```

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -71,15 +75,20 @@ def write_dataset(input: Channel,
-    :param write_dataset_processor: The python user defined function in write dataset operator. User can write their own logic here.
-    :param name: Name of the read_dataset operator.
+    Writes the dataset through the user-defined processor based on the metadata of dataset. The processor writes dataset
+    to external system and generates the :class:`~ai_flow.meta.ai_graph.ai_node.WriteDatasetNode` with
```
full class name should be fixed:
```
:class:`~ai_flow.meta.ai_graph.ai_node.WriteDatasetNode `" -> :class:`~ai_flow.ai_graph.ai_node.WriteDatasetNode
```

There are some wrong usage in the file as well. Maybe we can do a global replacement to fix the issue.

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -71,15 +75,20 @@ def write_dataset(input: Channel,
+                                      :class:`~ai_flow.meta.dataset_meta.DatasetMeta` or Text or int. The `dataset_info`
+                                      means the name of dataset when the type is Text and it means the id of dataset
+                                      when the type is int, which gets the metadata of dataset by the id.or name of
+                                      dataset at default.
```
See comments for `read_dataset` method. Maybe the description can be improved following the same format of the comments for `read_dataset`

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -103,15 +112,17 @@ def transform(input: Union[Channel, List[Channel]],
-    :return: Channel or Tuple[Channel]. It returns Channel When the output_num is 1 and returns Tuple[Channel] when
-             the output_num is bigger than 1, and the len(Tuple(Channel)) is output_num.
+    Transforms dataset through the user-defined processor based on the input :class:`~ai_flow.graph.channel.Channel`.
+    The processor generates the transforming :class:`~ai_flow.meta.ai_graph.ai_node.AINode`.
```
```
:class:`~ai_flow.meta.ai_graph.ai_node.AINode` -> :class:`~ai_flow.ai_graph.ai_node.AINode`

```

There are some wrong usage in the file as well. Maybe we can do a global replacement to fix the issue.

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -103,15 +112,17 @@ def transform(input: Union[Channel, List[Channel]],
+
+    :param input: List of the input :class:`~ai_flow.graph.channel.Channel` that contains the multiple channels from the
+                           operators which generate the data used to transform.
+    :param transform_processor: The user-defined processor composed of the logic that transforms.
```
`logic that transforms.` -> `transformation logic.` may be clearer

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -521,12 +564,13 @@ def action_on_job_status(job_name: Text,
+    :param job_name: The name of the job triggered.
+    :param upstream_job_name: The name of the upstream job.
+    :param upstream_job_status: The status of the upstream job.
+    :param action: The :class:`~ai_flow.workflow.control_edge.JobAction` acts on the source channel which value includes
```
` which value` ->  `whose value`

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -494,17 +537,17 @@ def action_on_model_version_event(job_name: Text,
+    :param job_name: The name of the job triggered.
+    :param dataset_name: The name of the dataset refer to a specified dataset.
+    :param namespace: The namespace of the event which uses the name of the project.
+    :param action: The :class:`~ai_flow.workflow.control_edge.JobAction` acts on the source channel which value includes
```
`which value` -> `whose value`

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -469,18 +512,18 @@ def action_on_model_version_event(job_name: Text,
-    :param job_name: The job name
-    :param model_name: Name of the model, refers to a specific model.
-    :return: None.
+    Add model version control dependency. It means the source channel will start when and only a new model version of
```
From my understanding, in our refractored API, we do not have `channel` concept in the `action_on_xx` API. Instead, we use the `job_name`. 

If we agree on this point, I think it is better to change `It means the source channel will start when and only a new model version of the specified model is updated in notification service.`  to `It means the job will take given action when the specified model is updated to given model_version_event_type. `

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -176,22 +193,27 @@ def predict(input: Union[Channel, List[Channel]],
-    :param output_num: The output number of the operator. The default value is 1.
-    :param name: Name of the predict operator.
-    :return: Channel or Tuple[Channel].
+    Predicts model through the user-defined processor with the information of model and model version based on the input
```
`Predicts model` -> `Predicts`

bgeng777 commented on flink-ai-flow/ai_flow/api/ops.py:
```diff
@@ -176,22 +191,27 @@ def predict(input: Union[Channel, List[Channel]],
+    :param prediction_processor: The user-defined processor composed of the logic that predicts model.
+    :param model_version_info: Information about the model version predicted by the processor. The type of model version
+                                                  could be :class:`~ai_flow.meta.model_meta.ModelVersionMeta` or Text(
+                                                  i.e. the version of model version). If the type is Text, this method
```
`i.e. the version of model version ` -> `i.e. the version field of ModelVersionMeta`

bgeng777 reviewed (approved):
LGTM.  Thanks a lot for the document refractor.

SteNicholas commented:
@bgeng777 , thanks for the detailed review. I have addressed all the above comments and squashed the commits to one commit. Please help to check this again.
